### PR TITLE
[CI] Speed up wheel commit validation check by 100x

### PR DESCRIPTION
**Pull request by can-anyscale**
## Why are these changes needed?

Speed up wheel commit validation check by 100x. Also hopefully will alleviate if not eliminate the 'Observed wheel commit () is not expected' issue (https://github.com/ray-project/ray/issues/32156) that has been creeping through many of ci/cd builds in our pipeline.

The existing code uses pipe to read from a rather large file (>50MB). Pipe however has buffer limit which by default in term of kb (https://man7.org/linux/man-pages/man7/pipe.7.html) so what we look for might not exist. We can fix this by tell unzip the exact file we are looking for. That file is pretty small so we should not hit buffer limit.

You might notice other surpises might still happen with this fix (e.g. many files that match ^__commit__). This sanity check goes back to 2 years ago by our veteran Kai (https://github.com/ray-project/ray/commit/234b015b426274d461a15345a4d4724a08bc5289) to sanity check issues with stale artifacts from previous builds or race conditions between builds. Further investigation on how builkite agent multi-tenant is setup might or might not simplify this logic further.

Signed-off-by: Cuong Nguyen <can@anyscale.com>
@can-anyscale

## Related issue number

#32156

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] CI Tests

### CODE REVIEW

**Comment by aslonnie** [comment]
linux pipe is a streaming setup, right? why the buffer size matters?

**Comment by aslonnie** [comment]
are you able to reproduce the issue and find out the root cause?

**Comment by can-anyscale** [comment]
@aslonnie good point, and you're absolutely correct, the stdin is stream; purely a theory but my best guess is streaming such a large stdin makes buffer and grep unpredictable, and hopefully no harm to  reduce the data usage a bit

I can find and download the problematic wheel (e.g. see artifacts and download cp39 in https://buildkite.com/ray-project/oss-ci-build-branch/builds/2791#0186ee52-dbab-4169-839c-f8eab80791be), but running the same grep command locally returns a proper commit, so the wheel is ok, but the empty commit return is not reproducible 

>  unzip -p ray-3.0.0.dev0-cp39-cp39-manylinux2014_aarch64.whl | grep "^__commit__" | awk '-F"' '{print $2}'
03a9d2166988b16b7cbf51dac0e6e586455b28d8

> unzip -p ray-3.0.0.dev0-cp39-cp39-manylinux2014_aarch64.whl "*/ray/__init__.py" | grep "^__commit__" | awk '-F"' '{print $2}'
03a9d2166988b16b7cbf51dac0e6e586455b28d8

The second command is also much (100x) faster

File: ci/ci.sh [hunk 1]
```diff
@@ -415,7 +415,7 @@ validate_wheels_commit_str() {
       continue
     fi
 
-    WHL_COMMIT=$(unzip -p "$whl" | grep "^__commit__" | awk -F'"' '{print $2}')
+    WHL_COMMIT=$(unzip -p "$whl" "*ray/__init__.py" | grep "^__commit__" | awk -F'"' '{print $2}')
 
     if [ "${WHL_COMMIT}" != "${EXPECTED_COMMIT}" ]; then
       echo "Error: Observed wheel commit (${WHL_COMMIT}) is not expected commit (${EXPECTED_COMMIT}). Aborting."
```
Comment (cadedaniel) [inline ci/ci.sh:421]: Can we add a log to print which wheel file it came from?

**Review by cadedaniel — APPROVED** [review]
I'm sus of the hypothesis that large files would cause grep to fail. But the speed up is great 😀 maybe change PR title to something like "speed up commit extraction from wheel"?

File: ci/ci.sh (hunk 1 above)
Comment (can-anyscale) [reply ci/ci.sh:421]: Absolutely!

**Comment by cadedaniel** [comment]
cc @jjyao

**Pull request merged**